.github/workflows/cibuildwheel.yml

```diff
@@ -94,10 +94,10 @@ jobs:
     runs-on: ${{ matrix.os }}
     env:
       CIBW_BEFORE_BUILD: >-
-        pip install certifi numpy>=1.25 &&
+        pip install certifi numpy &&
         rm -rf {package}/build
       CIBW_BEFORE_BUILD_WINDOWS: >-
-        pip install certifi delvewheel numpy>=1.25 &&
+        pip install certifi delvewheel numpy &&
```

Comment:
I forget now, but do we need NumPy in the pre-install now that it's in build requires?

Reply:
An early attempt here tried only changing this pin, only to find that it was using build isolation and therefore the wheel was built with 1.26 anyway. So I just removed the pin; it is in a CI job which will get the latest regardless, and doesn't actually affect what is used to build the wheel.

I also did a version which set `PIP_PRE` as an env var, which worked, but ultimately decided that changing pyproject.toml was appropriate, especially with the comment explaining the pins, which was taken from scipy's pyproject.toml.

Reply:
Well that's what I mean, I guess; if we are using build isolation, do we need `numpy` here at all?

Reply:
I think you are likely correct. I was leaning towards minimizing the difference, but I'll see if removing it is a problem... I guess there could be things to do with the delvewheel/other binary cleanup steps that may require numpy to be installed? 

Numpy has been installed here since #18233, which was right near the beginning of using cibw. It used to be pinned explicitly, then was `oldest-supported-numpy` until 1.25 allowed us to again use the latest version.

Reply:
actually, if the numpy dlls are needed for delvewheel (and the mac/linux equivalents that cibw uses by default), perhaps it is actually important for this _to_ be pinned to >=2.0.0rc1...

Reply:
It looks to me like the only dll added by any of these tools is msvc (on windows), so I'll go ahead and remove the numpy install and see if anything explodes.

Reply:
Yes, I think that's the only place that's needed; we definitely don't bundle any NumPy libraries.